examples/bridge-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,452 @@
+
+    strncpy(zclOctetStrBuf, value, sizeof(zclOctetStrBuf));
+    labelStruct.value = chip::ByteSpan(reinterpret_cast<uint8_t *>(&zclOctetStrBuf[0]), sizeof(zclOctetStrBuf));
+
+    emberAfCopyList(ZCL_FIXED_LABEL_CLUSTER_ID, am, true, buffer, reinterpret_cast<uint8_t *>(&labelStruct), 1);
+    emberAfCopyList(ZCL_FIXED_LABEL_CLUSTER_ID, am, true, buffer, reinterpret_cast<uint8_t *>(&listCount), 0);
+}
+
+EmberAfStatus HandleReadBridgedDeviceBasicAttribute(Device * dev, chip::AttributeId attributeId, uint8_t * buffer,
+                                                    uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "HandleReadBridgedDeviceBasicAttribute: attrId=%d, maxReadLength=%d", attributeId, maxReadLength);
+
+    if ((attributeId == ZCL_REACHABLE_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *buffer = dev->IsReachable() ? 1 : 0;
+    }
+    else if ((attributeId == ZCL_USER_LABEL_ATTRIBUTE_ID) && (maxReadLength == 32))
+    {
+        ToZclCharString(buffer, dev->GetName(), static_cast<uint8_t>(maxReadLength - 1));
+    }
+    else
+    {
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+EmberAfStatus HandleReadDescriptorAttribute(uint16_t endpointIndex, chip::AttributeId attributeId, uint8_t * buffer,
+                                            uint16_t maxReadLength)
+{
+    if ((maxReadLength <= kDescriptorAttributeArraySize) && (attributeId < kDescriptorAttributeCount))
+    {
+        memcpy(buffer, &gDescriptorAttrStorage[endpointIndex][attributeId][0], maxReadLength);
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus HandleReadFixedLabelAttribute(Device * dev, EmberAfAttributeMetadata * am, uint8_t * buffer, uint16_t maxReadLength)
+{
+    if ((am->attributeId == ZCL_LABEL_LIST_ATTRIBUTE_ID) && (maxReadLength <= kFixedLabelAttributeArraySize))
+    {
+        EncodeFixedLabel("room", dev->GetLocation(), buffer, maxReadLength, am);
+
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus HandleReadOnOffAttribute(Device * dev, chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "HandleReadOnOffAttribute: attrId=%d, maxReadLength=%d", attributeId, maxReadLength);
+
+    if ((attributeId == ZCL_ON_OFF_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *buffer = dev->IsOn() ? 1 : 0;
+    }
+    else
+    {
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+EmberAfStatus HandleWriteOnOffAttribute(Device * dev, chip::AttributeId attributeId, uint8_t * buffer)
+{
+    ChipLogProgress(DeviceLayer, "HandleWriteOnOffAttribute: attrId=%d", attributeId);
+
+    if ((attributeId == ZCL_ON_OFF_ATTRIBUTE_ID) && (dev->IsReachable()))
+    {
+        if (*buffer)
+        {
+            dev->SetOnOff(true);
```

Comment:
`dev->SetOnOff(*buffer !=  0);` could be shorter. I would also invert the condition to do an early 'return error' to reduce nesting. Code that reads `if (cond) { return ERR; }....`  seems easier to parse than having the return error at the tail.

```cpp
if (attributeId != ZCL_ON_OFF_ATTRIBUTE_ID) || !dev->IsReachable()) {
  return EMBER_ZCL_STATUS_FAILURE;
}

dev->SetOnOff(*buffer !=  0);
return EMBER_ZCL_STATUS_SUCCESS;
```

or you could use return macros for even shorter code.

Reply:
done